### ISSUE TITLE
Files opened through harpoon aren't added to the buffer list

### ISSUE THREAD

**Issue by polirritmico** [ThePrimeagen/harpoon#377]
Hi, great plugin! I've updated to version 2 but got this issue.

## The problem.

When opening files directly through Harpoon (added from a previous session), the files aren't added to the nvim buffers list.

## To reproduce:

0. Config:

```lua
local harpoon = require("harpoon")
harpoon:setup()
```

1. Test scenario:
```command
$ mkdir new_folder
$ cd new_folder
$ echo "file1 content" >> file1; echo "file2 content" >> file2; echo "file3 content" >> file3
$ nvim file1 file2 file3
```

The buffer list:

```
:buffers
1 %a   "file1"                        line 1
2      "file2"                        line 0
3      "file3"                        line 0
```

2. Add each file to the harpoon list:
```
:lua require("harpoon"):list():append()
:bn
:lua require("harpoon"):list():append()
:bn
:lua require("harpoon"):list():append()
```

3. Close and open a new nvim session:
```
:q
$ nvim
```

4. Go to any file through Harpoon:

```
:lua require("harpoon"):setup():list():select(2)
```

5. Check the buffer list:

```
:buffers

```

## Expected behaviour:

The output at step 5 should be:

```
:buffers
1 %a   "file2"                        line 1
```

Thanks!

--------------------------

## info

```command
$ nvim --version
nvim --version
NVIM v0.9.4
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/x86_64-pc-linux-gnu-gcc  -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fno-common -Wno-unused-result -Wimplicit-fallthrough -fdiagnostics-color=always -fstack-protector-strong -DUNIT_TESTING -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/usr/include/luajit-2.1 -I/usr/include -I/usr/include -I/var/tmp/portage/app-editors/neovim-0.9.4/work/neovim-0.9.4_build/src/nvim/auto -I/var/tmp/portage/app-editors/neovim-0.9.4/work/neovim-0.9.4_build/include -I/var/tmp/portage/app-editors/neovim-0.9.4/work/neovim-0.9.4_build/cmake.config -I/var/tmp/portage/app-editors/neovim-0.9.4/work/neovim-0.9.4/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include

archivo "vimrc" del sistema: "/etc/vim/sysinit.vim"
predefinido para $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

```
harpoon 0.02ms start
        dir    $XDG_DATA_HOME/nvim/lazy/harpoon
        url    https://github.com/ThePrimeagen/harpoon
        branch harpoon2
        commit c24c711

**Comment by ThePrimeagen** [comment]
i am going to get logging to also log out adding to lists because i don't think i can repro this one.  but the logging isn't quite ready yet, will be ready shortly

**Comment by ThePrimeagen** [comment]
ok, made some logger updates.  could you please update to the latest harpoon2 and perform the exact actions expected and then call `:lua require("harpoon").logger:show()` and copy that log over here?

we may need to iterate on this

**Comment by polirritmico** [comment]
> ok, made some logger updates. could you please update to the latest harpoon2 and perform the exact actions expected and then call `:lua require("harpoon").logger:show()` and copy that log over here?
> 
> we may need to iterate on this

Done:

```
config_default#select { context = { col = 0, row = 1 }, value = "file2" } __harpoon_files
```

**Comment by ThePrimeagen** [comment]
I think I know the problem and the logs don't help :)

**Comment by polirritmico** [comment]
Fixed by #392
